src/glewlwyd.c

```diff
@@ -1518,22 +1531,33 @@ void print_help(FILE * output) {
  * handles signal catch to exit properly when ^C is used for example
  * I don't like global variables but it looks fine to people who designed this
  */
-void exit_handler(int signal) {
-  if (signal == SIGQUIT || signal == SIGINT || signal == SIGTERM || signal == SIGHUP) {
-    y_log_message(Y_LOG_LEVEL_INFO, "Glewlwyd - Received close signal");
-  } else if (signal == SIGBUS) {
-    y_log_message(Y_LOG_LEVEL_ERROR, "Glewlwyd - Received bus error signal");
-    exit(1);
-  } else if (signal == SIGSEGV) {
-    y_log_message(Y_LOG_LEVEL_ERROR, "Glewlwyd - Received segmentation fault signal");
-    exit(1);
-  } else if (signal == SIGILL) {
-    y_log_message(Y_LOG_LEVEL_ERROR, "Glewlwyd - Received illegal instruction signal");
-    exit(1);
-  }
-  pthread_mutex_lock(&global_handler_close_lock);
-  pthread_cond_signal(&global_handler_close_cond);
-  pthread_mutex_unlock(&global_handler_close_lock);
+void* signal_thread(void *arg) {
+  sigset_t *sigs = arg;
+  int res, signum;
+
+  while (1) {
```

Comment:
I'm not sure putting an infinite loop here is relevant. If I understand this function correctly, it waits until an expected signal is sent, then broadcasts a closing signal with `global_handler_close_cond` or exits if the signal is bad.
In any case, when the signal is received, it will close the program, or there may be other `signum` values?

If we need to keep the `while (1) {`, then I'd add a `usleep(50)` right before the end loop to avoid having the CPU at 100%.

Reply:
Practically there won't be other signum values, because we only wait for signals in `sigs` and we check for all of them manually. All signals we receive will end up exiting the thread (and program) in some way, so the infinite loop is not really needed.

It would be useful if there were other signals that would be handled without exiting (typical would be "reload configuration on SIGHUP" or something like that). Then we really should continue to wait for the more signal that might come. I'll add a log warning for any unexpected received signals, even though it should never happen currently. It's a bit more resilient in case more signals are added later. But the loop could also be removed if you prefer it.

I don't think a sleep is really necessary: if there are signals generated at a rate that takes 100% CPU to process, than the real problem is wherever those signals are generated. The thread processing them can't really be blamed that many of them are coming in.

Reply:
Then I would remove the while loop